torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2533,7 +2533,8 @@ def maybe_non_contig(tensor):
                 arg = arg.double()
             if arg.dtype == torch.cfloat:
                 arg = arg.to(torch.cdouble)
-            if arg.is_complex() != dtype.is_complex:
+            # enable torch.int64 argument representing indices
+            if (arg.is_complex() != dtype.is_complex) and (arg.dtype != torch.int64):
```

Comment:
@anjali411 , could you please tell whether it is fine to have this for now?

Reply:
Mmm I understand why you considered adding this, but I don't think we should add this. would you mind adding an `OpInfo` test for `torch.index_fill`?